arangod/Aql/EngineInfoContainerDBServer.cpp

```diff
@@ -523,7 +615,11 @@ void EngineInfoContainerDBServer::DBServerInfo::injectQueryOptions(
 std::map<ServerID, EngineInfoContainerDBServer::DBServerInfo>
 EngineInfoContainerDBServer::createDBServerMapping(
     std::unordered_set<ShardID>& lockedShards) const {
-  auto ci = ClusterInfo::instance();
+  auto* ci = ClusterInfo::instance();
+
+  if (!ci) {
```

Comment:
why not assert on != nullptr ?

Reply:
assertion doesn't work for release, so this code at least isn't going to cause segfault. I believe we have to consider possibility to return reference instead of pointer, if `ClusterInfo` is not intended to be equal `nullptr` of course.

Reply:
I think it can't be a nullptr after the startup finishes